functions/Get-DbaAgListener.ps1

```diff
@@ -0,0 +1,103 @@
+			Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+			License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+		.LINK
+			https://dbatools.io/Get-DbaAgListener
+
+		.EXAMPLE
+			Get-DbaAgListener -SqlInstance sqlserver2014a
+
+			Returns basic information on all the databases in each Availability Group found on sqlserver2014a
+
+		.EXAMPLE
+			Get-DbaAgListener -SqlInstance sqlserver2014a -AvailabilityGroup AG-a
+
+			Returns basic information on all the databases in the Availability Group AG-a on sqlserver2014a
+
+		.EXAMPLE
+			Get-DbaAgListener -SqlInstance sqlserver2014a -AvailabilityGroup AG-a -Database AG-Database
+
+			Returns basic information on the database AG-Database found in the Availability Group AG-a on server sqlserver2014a
+	#>
+	[CmdletBinding()]
+	param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[PSCredential][System.Management.Automation.CredentialAttribute()]
+		$SqlCredential,
+		[parameter(ValueFromPipeline = $true)]
+		[object[]]$AvailabilityGroup,
+		[object[]]$Database,
+		[switch]$Silent
+	)
+
+	process {
+		foreach ($serverName in $SqlInstance) {
+			try {
+				$server = Connect-SqlInstance -SqlInstance $serverName -SqlCredential $SqlCredential -MinimumVersion 11
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+			}
+
+			if ($server.IsHadrEnabled -eq $false) {
+				Stop-Function -Message "Availability Group (HADR) is not configured for the instance: $serverName." -Target $serverName -Continue
+			}
+
+			$ags = $server.AvailabilityGroups
+			if ($AvailabilityGroup) {
+				$ags = $ags | Where-Object Name -in $AvailabilityGroup
+			}
+
+			foreach ($ag in $ags) {
+				Add-Member -Force -InputObject $ag -MemberType NoteProperty -Name ComputerName -value $server.NetName
+				Add-Member -Force -InputObject $ag -MemberType NoteProperty -Name InstanceName -value $server.ServiceName
+				Add-Member -Force -InputObject $ag -MemberType NoteProperty -Name SqlInstance -value $server.DomainInstanceName
+				
+				$defaults = 'Name as AvailabilityGroup','ComputerName','InstanceName','SqlInstance','ClusterIPConfiguration','PortNumber','AvailabilityGroupListenerIPAddresses','AGListenerName'
+				$AvailabilityGroup = Get-DbaAvailabilityGroup $SqlInstance
+				$listener = $AvailabilityGroup.AvailabilityGroupListeners
+				Add-Member -Force -InputObject $ag -MemberType NoteProperty -Name ClusterIPConfiguration -value $listener.ClusterIPConfiguration
+				Add-Member -Force -InputObject $ag -MemberType NoteProperty -Name PortNumber -value $listener.PortNumber
+				Add-Member -Force -InputObject $ag -MemberType NoteProperty -Name AvailabilityGroupListenerIPAddresses -value $listener.AvailabilityGroupListenerIPAddresses
+				Add-Member -Force -InputObject $ag -MemberType NoteProperty -Name AGListenerName -value $listener.Name
```

Comment:
We need to return the `$listener` object and not the full Availability Group object (`$ag`).

So you can adjust this in your `$defaults` as to what properties we want to spit out and then pass the `$listener` as the input object for `Select-DefaultView`.

Small example:
```powershell
$defaults = 'Parent as AvailabilityGroupName','Name as ListenerName','PortNumber'...
Select-DefaultView -InputObject $listener -Property $defaults
```